contracts/events/TopicEvent.sol

```diff
@@ -299,54 +292,46 @@ contract TopicEvent is ITopicEvent, BaseContract, Ownable {
-    * @return The amount of QTUM tokens won.
+    * @notice Calculates the BOT and QTUM tokens won based on the sender's contributions.
+    * @return The amount of BOT and QTUM tokens won.
     */
-    function calculateQtumContributorWinnings() 
+    function calculateWinnings()
         public 
         view
         inCollectionStatus()
-        returns (uint256)  
+        returns (uint256, uint256)  
     {
-        uint256 senderContribution = balances[resultIndex].bets[msg.sender];
+        uint256 votes = balances[resultIndex].votes[msg.sender];
+        uint256 bets = balances[resultIndex].bets[msg.sender];
+        if (votes == 0 && bets == 0) {
+            return (0, 0);
+        }
+
+        // Calculate QTUM winnings
         uint256 winnersTotal = balances[resultIndex].totalBets;
-        uint256 losersTotalMinusCut = 0;
+        uint256 losersTotal = 0;
         for (uint8 i = 0; i < numOfResults; i++) {
             if (i != resultIndex) {
-                losersTotalMinusCut = losersTotalMinusCut.add(balances[i].totalBets);
+                losersTotal = losersTotal.add(balances[i].totalBets);
             }
         }
-        losersTotalMinusCut = losersTotalMinusCut.mul(90).div(100);
-        uint256 senderContributionMinusCut = senderContribution.mul(90).div(100);
-        return senderContribution.mul(losersTotalMinusCut).div(winnersTotal).add(senderContributionMinusCut);
-    }
-
-    /*
-    * @notice Calculates the QTUM and BOT tokens won based on the sender's BOT contributions.
-    * @return The amount of QTUM and BOT tokens won.
-    */
-    function calculateBotContributorWinnings()
-        public
-        view
-        inCollectionStatus()
-        returns (uint256, uint256) 
-    {
-        // Calculate BOT won
-        uint256 senderContribution = balances[resultIndex].votes[msg.sender];
-        uint256 winnersTotal = balances[resultIndex].totalVotes;
-        uint256 losersTotal = 0;
-        for (uint8 i = 0; i < numOfResults; i++) {
+        uint256 rewardQtum = uint256(QTUM_PERCENTAGE).mul(losersTotal).div(100);
+        losersTotal = losersTotal.sub(rewardQtum);
+        uint256 qtumWon = bets.mul(losersTotal).div(winnersTotal).add(bets);
+
+        // Calculate BOT winnings
+        winnersTotal = balances[resultIndex].totalVotes;
+        losersTotal = 0;
+        for (i = 0; i < numOfResults; i++) {
             if (i != resultIndex) {
                 losersTotal = losersTotal.add(balances[i].totalVotes);
             }
         }
-        uint256 botWon = senderContribution.mul(losersTotal).div(winnersTotal).add(senderContribution);
-
-        // Calculate QTUM won
-        uint256 qtumReward = totalQtumValue.mul(10).div(100);
-        uint256 qtumWon = senderContribution.mul(qtumReward).div(winnersTotal);
+        uint256 botWon = votes.mul(losersTotal).div(winnersTotal).add(votes);
+        uint256 rewardWon = votes.mul(rewardQtum).div(winnersTotal);
+        qtumWon = qtumWon.add(rewardWon);
```

Comment:
The logic looks good to me. Sorry for keeping this PR hanging for long.

Reply:
no worries. better to get it right the first time.